fifo.c

```diff
@@ -2,82 +2,317 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <time.h>
 #include <sys/time.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <stdbool.h>
 
+#define SECONDS_PER_MINUTE 60
+#define SIZE_OF_USED_BUFFER 24    /* we put a sequence number and a timestamp at the beginning
+                                  of the buffer */
+#define ARRAY_BOOST 20000         /* The size of the array initially and how much to increase it's
+                                  size as it grows */  
+
+/* Author:   Matt Currier
+ * 
+ * This code base is from :
+ * https://openbenchmarking.org/test/pts/ipc-benchmark ->
```

Comment:
@mcurrier2 Do you have code here actually taken from the openbenchmarking.org code? This could have license implications, so we need to know.